src/App.tsx

```diff
@@ -15,48 +17,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Default = '',
+  Name = 'alphabeticaly',
+  Length = 'length',
+}
+
+interface FunctionArguments {
+  sortField : string,
+  reversed: boolean,
+}
+
+function prepareGoods(goods: string[],
+  { sortField, reversed }: FunctionArguments): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.Name:
+          return good1.localeCompare(good2);
+
+        case SortType.Length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    return preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.Default);
+  const [reversed, setReversed] = useState(false);
+
+  const reset = () => {
+    setSortField(SortType.Default);
+    setReversed(false);
+  };
+
+  const visibleGoods = prepareGoods(goodsFromServer, { sortField, reversed });
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={
+            cn('button', 'is-info',
+              { 'is-light': sortField !== SortType.Name })
+          }
+          onClick={() => setSortField(SortType.Name)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={
+            cn('button', 'is-success',
+              { 'is-light': sortField !== SortType.Length })
+          }
+          onClick={() => setSortField(SortType.Length)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={
+            cn('button', 'is-warning',
+              { 'is-light': !reversed })
+          }
+          onClick={() => setReversed(!reversed)}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortField || reversed) && (
```

Comment:
Create a variable for `sortField || reversed` condition to avoid mixing different logical operators